admin_organization.go

```diff
@@ -0,0 +1,169 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ AdminOrganizations = (*adminOrganizations)(nil)
+
+// AdminOrganizations describes all of the admin organization related methods that the Terraform
+// Enterprise API supports. Note that admin settings are only available in Terraform Enterprise.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/admin/organizations.html
+// Module sharing docs: https://www.terraform.io/docs/cloud/api/admin/module-sharing.html
+type AdminOrganizations interface {
+
+	// List the module sharing partnerships that an organization has
+	ListModuleConsumers(ctx context.Context, organization string, options OrganizationListOptions) (*OrganizationList, error)
+
+	// Update the module sharing consumers that an organization has
+	UpdateModuleConsumers(ctx context.Context, organization string, consumers ModuleConsumers) error
+
+	// Read attributes of an existing organization via admin API.
+	Read(ctx context.Context, organization string) (*AdminOrganization, error)
+
+	// Update attributes of an existing organization via admin API.
+	Update(ctx context.Context, organization string, options AdminOrganizationUpdateOptions) (*AdminOrganization, error)
+
+	// Delete an organization by its name via admin API
+	Delete(ctx context.Context, organization string) error
+}
+
+// adminOrganizations implements AdminOrganizations.
+type adminOrganizations struct {
+	client *Client
+}
+
+// ModuleConsumers is a typed list of organization names to represent the
+// list of module consumers for updates
+type ModuleConsumers []*string
+
+// modulePartnershipUpdateOption represents the option for updating an organization's module sharing partnerships
+// this is private as it's used internally
+type modulePartnershipUpdateOption struct {
+	ID *string `jsonapi:"primary,organizations"`
+}
+
+// AdminOrganization represents a Terraform Enterprise organization returned from the Admin API.
+type AdminOrganization struct {
+	Name string `jsonapi:"primary,organizations"`
+
+	AccessBetaTools                  bool   `jsonapi:"attr,access-beta-tools"`
+	ExternalID                       string `jsonapi:"attr,external-id"`
+	GlobalModuleSharing              bool   `jsonapi:"attr,global-module-sharing"`
+	IsDisabled                       bool   `jsonapi:"attr,is-disabled"`
+	NotificationEmail                string `jsonapi:"attr,notification-email"`
+	SsoEnabled                       bool   `jsonapi:"attr,sso-enabled"`
+	TerraformBuildWorkerApplyTimeout string `jsonapi:"attr,terraform-build-worker-apply-timeout"`
```

Comment:
I see that there is also a `terraform-worker-sudo-enabled` field returned for organizations (looking at[ docs](https://www.terraform.io/docs/cloud/api/admin/organizations.html#sample-response))- although it looks like you can't update it. Is it something that should be added to the struct?

Reply:
Added this in there.